components/ws-manager-api/typescript/src/client-provider-source.ts

```diff
@@ -27,8 +27,15 @@ export class WorkspaceManagerClientProviderEnvSource implements WorkspaceManager
         return this.clusters.find((m) => m.name === name && m.applicationCluster === applicationCluster);
     }
 
-    public async getAllWorkspaceClusters(applicationCluster: string): Promise<WorkspaceClusterWoTLS[]> {
-        return this.clusters.filter((m) => m.applicationCluster === applicationCluster) ?? [];
+    public async getAllWorkspaceClusters(
+        applicationCluster: string,
+        region?: string,
+    ): Promise<WorkspaceClusterWoTLS[]> {
+        return (
+            this.clusters.filter(
+                (m) => m.applicationCluster === applicationCluster && (!region || m.region === region),
+            ) ?? []
```

Comment:
Why do we need the coalesce here? `this.clusters.filter` returns the empty list already, if the values do not match - we are guaranteed to get back a list (even if emtpy)

Reply:
I don't know why it was there. will remove